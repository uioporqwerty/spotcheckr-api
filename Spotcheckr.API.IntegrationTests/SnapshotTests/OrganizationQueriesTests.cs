using System.Collections.Generic;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Spotcheckr.Data.Repositories;
using Spotcheckr.Domain;
using Xunit;

namespace Spotcheckr.API.IntegrationTests.SnapshotTests
{
	public class OrganizationQueriesTests : BaseSnapshotTest
	{
		private readonly IUnitOfWork UnitOfWork;

		public OrganizationQueriesTests()
		{
			UnitOfWork = ServiceProvider.GetRequiredService<IUnitOfWork>();
		}

		[Fact]
		public async void GetOrganizations_WithGet_ReturnsOrganizations()
		{
			var organizations = new List<Organization>
			{
				new Organization("CPT", "Some Name"),
				new Organization("CPP", "Some other name")
			};
			UnitOfWork.Organizations.AddRange(organizations);
			UnitOfWork.Complete();

			var executor = await GetRequestExecutorAsync();
			var result = await executor.ExecuteAsync(@"
				query {
					organizations {
						abbreviation
						name
					}
				}
			");
			result.ToJson().MatchSnapshot();
		}
	}
}
