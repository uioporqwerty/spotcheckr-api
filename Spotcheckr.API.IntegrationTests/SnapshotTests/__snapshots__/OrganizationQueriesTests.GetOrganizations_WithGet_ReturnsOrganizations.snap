{
  "data": {
    "organizations": [
      {
        "abbreviation": "CPT",
        "name": "Some Name"
      },
      {
        "abbreviation": "CPP",
        "name": "Some other name"
      }
    ]
  }
}
